hibernate-core/src/main/java/org/hibernate/internal/util/LockModeConverter.java

```diff
@@ -28,27 +28,26 @@ private LockModeConverter() {
 	 * @return The JPA {@link LockModeType}
 	 */
 	public static LockModeType convertToLockModeType(LockMode lockMode) {
-		if ( lockMode == LockMode.NONE ) {
-			return LockModeType.NONE;
-		}
-		else if ( lockMode == LockMode.OPTIMISTIC || lockMode == LockMode.READ ) {
-			return LockModeType.OPTIMISTIC;
-		}
-		else if ( lockMode == LockMode.OPTIMISTIC_FORCE_INCREMENT || lockMode == LockMode.WRITE ) {
-			return LockModeType.OPTIMISTIC_FORCE_INCREMENT;
-		}
-		else if ( lockMode == LockMode.PESSIMISTIC_READ ) {
-			return LockModeType.PESSIMISTIC_READ;
-		}
-		else if ( lockMode == LockMode.PESSIMISTIC_WRITE
-				|| lockMode == LockMode.UPGRADE_NOWAIT
-				|| lockMode == LockMode.UPGRADE_SKIPLOCKED) {
-			return LockModeType.PESSIMISTIC_WRITE;
-		}
-		else if ( lockMode == LockMode.PESSIMISTIC_FORCE_INCREMENT ) {
-			return LockModeType.PESSIMISTIC_FORCE_INCREMENT;
+		switch (lockMode) {
+			case NONE:
+			case READ: // no exact equivalent in JPA
+				return LockModeType.NONE;
+			case OPTIMISTIC:
+				return LockModeType.OPTIMISTIC;
+			case WRITE: // no exact equivalent in JPA
```

Comment:
Hey @gavinking, maybe I'm missing something, but the semantics of `WRITE` is that a row for an entity was inserted/updated, which implies the row is exclusive locked, so maybe the mapping to `PESSIMISTIC_WRITE` is more appropriate?

Reply:
Yeah it's tricky, and I'm really not sure. Nothing really works 100%. 

- I don't think the mapping to `PESSIMISTIC_WRITE` is all that natural, since that's just an upgrade lock, not an actual write lock.
- You could definitely make a case for mapping to `PESSIMISTIC_FORCE_INCREMENT`, since that indeed obtains an immediate write lock.

Well, wait....

Actually I think i should definitely make the change to map it to `PESSIMISTIC_FORCE_INCREMENT`.

-  Previously, `WRITE` for some reason had a numeric level similar to `OPTIMISTIC_FORCE_INCREMENT`, which I guess is why I had it mapping to `OPTIMISTIC_FORCE_INCREMENT`. 
- But I changed that in this commit and gave it the same level as `PESSIMISTIC_FORCE_INCREMENT`.

So let me indeed make that change.